Feature/ERC20-token-transfer-across-chains (#1572)
iotaledger/iota-wiki: docs/build/isc/v1.0.0-rc.6/docs/how-tos/send-ERC20-across-chains.md

### Using IOTA's Implementation of [Utilities for LayerZero OFT V2](https://github.com/iota-community/layerzero-oft-v2-utils)

The Utilities for LayerZero OFT V2 facilitate cross-chain sending of ERC20 tokens between a source chain (e.g., Sepolia or ShimmerEVM testnet) and a destination chain (e.g., BNB testnet or IOTA EVM testnet).

:::tip
Further information: https://docs.layerzero.network/v2/developers/evm/gas-settings/options#option-types.
:::

### Procedure to Send Tokens from Source Chain to Destination Chain and Vice Versa

For existing ERC20 tokens that involve both the OFTAdapter contract (on the source chain) and the OFT contract (on the destination chain), the token sending procedure is as follows:

1. **Approve Tokens**: The sender approves their ERC20 tokens for the OFTAdapter contract.
2. **Estimate Fee**: The sender calls the function `quoteSend()` of the OFTAdapter contract to estimate the cross-chain fee to be paid in native tokens on the source chain.
3. **Send Tokens**: The sender calls the function `send()` of the OFTAdapter contract to transfer tokens from the source chain to the destination chain.

> 4. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library [@layerzerolabs/scan-client](https://www.npmjs.com/package/@layerzerolabs/scan-client#example-usage).

Reviewer: ```suggestion
4. **(Optional) Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library [@layerzerolabs/scan-client](https://www.npmjs.com/package/@layerzerolabs/scan-client#example-usage).
```

To send back the OFT-wrapped tokens on the destination chain to the source chain, the procedure is similar, except that the approval step is not needed:

1. **Estimate Fee**: The sender calls the function `quoteSend()` of the OFT contract to estimate the cross-chain fee to be paid in native tokens on the sender chain.
2. **Send Tokens**: The sender calls the function `send()` of the OFT contract to transfer tokens from the destination chain back to the source chain.

> 3. **Optional - Wait for Finalization**: The sender can wait for transaction finalization on the destination chain using the library `@layerzerolabs/scan-client`.

Author: Sure Lucas, I have added code samples for all the above points required.

> #### Sample Solidity code for OFTAdater and OFT contracts in folder `contracts-standard`

Reviewer: ```suggestion
## Sample Solidity code for OFT Adapter and OFT contracts in the folder `contracts-standard`
```

> - Scripts for:
>   - Deploy OFTAdapter and OFT contracts
>   - Set trusted peer
>   - Set enforced options
>   - Send tokens from source chain to destination chain and vice versa

Reviewer: ```suggestion
The [contracts-standard](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-standard) contains scripts to:

- Deploy the OFT Adapter and OFT contracts.
- Set your trusted peers.
- Set enforced options.
- Send tokens from the source chain to the destination chain, and vice versa.
```

## Installation
```
$ yarn
```

> ## Compile contracts

Reviewer: ```suggestion
## Compile the Contracts
```

> - In the standard implementation for ERC20, copy the folder `contracts-standard`to `contracts` and `contracts-wiota` to `contracts` for custom implementation. Then, run the command:

Reviewer: ```suggestion
If you want to use the standard implementation for ERC20, copy the [`contracts-standard`](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-standard) folder to `contracts`. If you want to use a custom implementation, copy the  [`contracts-wiota`](https://github.com/iota-community/layerzero-oft-v2-utils/tree/main/contracts-wiota) to `contracts`. Then, run the following command to compile the contracts:
```

```
  $ yarn compile
``` 

## Configuration

The config is specified in the template file `.env.example` that needs to be copied to another file `.env`.

> 
> ## Deploy contracts

Reviewer: ```suggestion
### Deploy the Contracts
```

### Deploy OFTAdapter on source chain (e.g. Sepolia)

The OFT Adapter facilitates the expansion of an existing token to any supported blockchain as a native token, maintaining a unified global supply and inheriting all features of the OFT Standard. This intermediary contract manages the sending and receiving of pre-deployed tokens.

For instance, when an ERC20 token is transferred from the source chain (Chain A), it gets locked in the OFT Adapter. Consequently, a corresponding token is minted on the destination chain (Chain B) through the paired OFT Contract.

```
$ yarn deploy-oft-adapter-sepolia
```

Log output :

```
$ npx hardhat run scripts/deploy_oft_adapter.ts --network sepolia
Deployed MyOFTAdapter contract address: 0x4daa81978576cB91a2e1919960e90e46c2a6D586
Done in 6.67s.
```

### Deploy OFT on destination chain (e.g. BNB testnet)

```
$ yarn deploy-oft-bnb-testnet
``` 

Log output :

```
$ npx hardhat run scripts/deploy_oft.ts --network bnbTestnet
Deployed MyOFT contract address: 0xCc337C2e69F4Eb8EaBcf632a1fC5B8F729dC47F1
Done in 6.68s.
```

## Set trusted peer (Optinal)

> ### On OFTAdapter (source chain, e.g. Sepolia)

Reviewer: ```suggestion
### On the OFT Adapter
```
